substrate/runtime/treasury/src/lib.rs

```diff
@@ -278,9 +278,13 @@ impl<T: Trait> Module<T> {
 	}
 }
 
-impl<T: Trait> OnMinted<T::Balance> for Module<T> {
-	fn on_minted(b: T::Balance) {
-		<Pot<T>>::put(Self::pot() + b);
+impl<T: Trait> OnDilution<T::Balance> for Module<T> {
+	fn on_dilution(minted: T::Balance, portion: T::Balance) {
+		if !minted.is_zero() && !portion.is_zero() {
```

Comment:
a comment here describing that extra tokens are minted to the treasury whenever new balances are minted would be useful